sdk/lib/drivers/sound/mmebuddy/wave/streaming.c

```diff
@@ -203,9 +209,24 @@ CompleteIO(
         }
 		else
 		{
-            /* Partially completed */
-            HdrExtension->BytesCompleted += dwNumberOfBytesTransferred;
-            SND_TRACE(L"%d/%d bytes of wavehdr completed\n", HdrExtension->BytesCompleted, WaveHdr->dwBufferLength);
+            /* Do we loop a header? */
+            if (HdrExtension->BytesCommitted == WaveHdr->dwBufferLength &&
+                !SoundOverlapped->PerformCompletion)
+            {
+                /* Reset amount of bytes and decrement loop count, to play next iteration */
+                SND_TRACE(L"Looping the header, remaining loops %d", SoundDeviceInstance->LoopsRemaining);
+                HdrExtension->BytesCommitted = 0;
+                SND_ASSERT(SoundDeviceInstance->LoopsRemaining > 0);
+                if (SoundDeviceInstance->LoopsRemaining != 0xFFFFFFFF)
```

Comment:
We cannot use the `INFINITE` define in this file?

Reply:
Yes, seems we can. At least it compiles successfully with it. So will use it instead.

Reply:
Done.